test/unit/dom/wait_for_test.dart

```diff
@@ -201,6 +201,16 @@ void main() {
                 hasToStringValue(contains('The callback must return a non-null Element.')),
               )));
         }, timeout: asyncQueryTestTimeout);
+
+        test('throws a timeout error based on a provided waitForElementToBeRemoved() timeout duration', () async {
+          expect(
+              () => rtl.waitForElementToBeRemoved(() => elementThatWillBeRemovedAfterDelay, container: view.container, 
+                timeout: Duration(milliseconds: 1)),
+              throwsA(allOf(
+                isA<TimeoutException>(),
+                hasToStringValue(contains('The element returned from the callback was still present in the container after 1ms:')),
+              )));
+        }, timeout: asyncQueryTestTimeout);
```

Comment:
This test is more of a sanity check. Since `waitForElementToBeRemoved` and `waitFor` share the same timeout duration, I'm testing that `waitForElementToBeRemoved` throws an error with the timeout value it's explicitly given (on line 208). 

This test works _even without_ our default timeout added in `wait_for.dart`, but my goal was to ensure confidence the thrown error uses the timeout value in `waitForElementToBeRemoved`.

We could also arbitrarily set the default timeout in `waitForElementToBeRemoved` lower than the default in `waitFor`, but that seems unnecessary.

Open to feedback on this one. What's another way to approach this? Gonna think more about it over the weekend. The brain fog is real

Reply:
I think that test looks good! 

I think if you also wanted to add a test that _would_ fail without the default timeout you added, then you could add another test that doesn't pass in a `timeout` arg to `waitForElementToBeRemoved` and verifies that the error message has the `config.asyncUtilTimeout` value rather than the "can't read properties of null" error that used to be happening because there was no default `timeout`. (this might also require running the test on a element that will never be removed from the document rather than one that has a delayed removal)

Reply:
+1 to Sydney! This test looks good, and I think Sydney's suggestions make a lot of sense as well. Also, +1 to having a great weekend!